pkg/util/installinfo/install_info_test.go

```diff
@@ -176,3 +176,116 @@ install_method:
+  tool_version: install_script
+  installer_version: install_script-x.x.x
+`,
+			version:   "1",
+			timestamp: time.Date(2022, 4, 12, 7, 10, 58, 1234, time.UTC),
+			want:      `{"entries":[{"version":"1","timestamp":"2022-04-12T07:10:58.000001234Z","install_method":{"tool":"install_script","tool_version":"install_script","installer_version":"install_script-x.x.x"}}]}`,
+		},
+		{
+			name:               "version in new entry is same as the last entry",
+			versionHistoryFile: `{"entries":[{"version":"1","timestamp":"2022-04-07T14:24:58.152534935Z"}]}`,
+			installInfoFile: `
+---
+install_method:
+  tool: install_script
+  tool_version: install_script
+  installer_version: install_script-x.x.x
+`,
+			version:   "1",
+			timestamp: time.Date(2022, 4, 12, 7, 10, 58, 1234, time.UTC),
+			want:      `{"entries":[{"version":"1","timestamp":"2022-04-07T14:24:58.152534935Z"}]}`,
+		},
+		{
+			name:               "version and timestamp of the new entry is empty",
+			versionHistoryFile: `{"entries":[{"version":"1","timestamp":"2022-04-07T14:24:58.152534935Z"}]}`,
+			installInfoFile: `
+---
+install_method:
+  tool: install_script
+  tool_version: install_script
+  installer_version: install_script-x.x.x
+`,
+			// we still need a version to be able to write a new entry
+			want: `{"entries":[{"version":"1","timestamp":"2022-04-07T14:24:58.152534935Z"}]}`,
+		},
+
+		{
+			name:               "existing version-history.json in invalid JSON",
+			versionHistoryFile: `{"entries":[{"1","timestamp":"2022-04-07T14:24:58.152534935Z"}]}`,
+			installInfoFile: `
+---
+install_method:
+  tool: install_script
+  tool_version: install_script
+  installer_version: install_script-x.x.x
+`,
+			version:   "2",
+			timestamp: time.Date(2022, 4, 12, 7, 10, 58, 1234, time.UTC),
+			want:      `{"entries":[{"version":"2","timestamp":"2022-04-12T07:10:58.000001234Z","install_method":{"tool":"install_script","tool_version":"install_script","installer_version":"install_script-x.x.x"}}]}`,
+		},
+		{
+			name:               "install_info in invalid YAML",
+			versionHistoryFile: `{"entries":[{"version":"1","timestamp":"2022-04-07T14:24:58.152534935Z"}]}`,
+			installInfoFile: `
+---
+install_method:
+  tool: install_script tool_version: install_script
+  installer_version: install_script-x.x.x
+`,
+			version:   "2",
+			timestamp: time.Date(2022, 4, 12, 7, 10, 58, 1234, time.UTC),
+			// now we have the install info through env vars, so the second version is populated
+			want: `{"entries":[{"version":"1","timestamp":"2022-04-07T14:24:58.152534935Z","install_method":{"tool":"","tool_version":"","installer_version":""}},{"version":"2","timestamp":"2022-04-12T07:10:58.000001234Z","install_method":{"tool":"install_script","tool_version":"install_script","installer_version":"install_script-x.x.x"}}]}`,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			vh, err := os.CreateTemp("", "version-history.json")
+			assert.Equal(t, nil, err)
```

Comment:
We should `require` here as there is no point in continuing the test if we failed to create the file.

```suggestion
			require.NoError(t, err)
```